docs/the-new-architecture/pillars-fabric-components.md

```diff
@@ -3,17 +3,857 @@ id: pillars-fabric-components
 title: Fabric Components
 ---
 
-This section contains a high-level introduction to Fabric components. It provides enough context to understand when a Fabric component is needed and how it roughly works.
-It points to the [Renderer](https://reactnative.dev/architecture/fabric-renderer) section of the [Architecture](https://reactnative.dev/architecture/overview) tab for a deep dive into the technical details.
+import Tabs from '@theme/Tabs'; import TabItem from '@theme/TabItem'; import constants from '@site/core/TabsConstants';
 
-This section must have a warning that it works only with the new architecture enabled. It points to the [migration section](../new-architecture-intro).
+A Fabric Component is a UI component rendered on the screen using the [Fabric Renderer](https://reactnative.dev/architecture/fabric-renderer).
+
+Using Fabric Components instead of Native Components allows us to reap all the [benefits](./why) of the **New Architecture**. Specifically, it enables to use JSI to efficiently connect the Native UI code with JavaScript.
+
+A Fabric Component is created starting from a **JavaScript specification**. Then [**CodeGen**](./pillars-codegen) creates some C++ code, integrated in the platform native layer and shared among all the React Native platforms. This C++ code is boilerplate code that component-specific logic needs to import and use to be properly adopted by React Native. After the component-specific logic is connected with the generated code, the component can be integrated in the app.
```

Comment:
This rephrasing looks pretty good! I have just a few smaller questions:

* "integrated in the platform native layer and shared among all the React Native platforms": I also took a stab at rephrasing a similar idea in the TM guide, let me know what you think: https://github.com/facebook/react-native-website/pull/3168/files#diff-ba6eb9aa37edbf22f752c4267cb34a52ff888f9a12d1795335fdfaf6dfc61f05R13.
  * I am thinking "integrated in the platform native layer" and "shared among all the RN platforms" might sound like the same thing to some readers (although I know what you mean) -- they might both sound like "C++ code that the native language you're used to seeing on each platform can use"). I think to disambiguate the two ideas here we might want a more specific phrasing than "integrate".
* I think "that component-specific logic needs to import and use to be properly adopted by React Native" still has some ambiguity for the reader. What is "component-specific logic"? What does "properly adopted" mean? Brainstorming a few ideas for other phrasing:
  * "This C++ code is scaffolding that your native component logic needs to utilize to work with the New Architecture"
  * "Then Codegen creates some C++ code, generating logic and interfaces that will help bridge your JavaScript code to the Native Platform. You can either add more C++ code or integrate the generated code with your Native Platform language of choice." (Also replaces the sentence before).
* nit: "boilerplate code": this is perfectly fine, but I really liked @cortinico's suggestion of "scaffolding code" and used it a few times in the TM guide, so we might want to consider being consistent about that

Reply:
Thanks Lizzi! I used your list to give consistency between the two guides. I then rewrote the paragraph on the Codegen, simplifying it a little bit and adding some examples to better explain the idea.